posts/2024-02-08-interviews-anirban_chetia/index.qmd

```diff
@@ -0,0 +1,161 @@
+
+I was assigned by Toby to do these interviews and serve as the EL (Entrepreneurial Lead) for the `data.table` team (which aside from this, involved tasks such as making and giving various presentations). Having successfully completed the program and conducted the interviews, it's time to share the insights I gathered from these interviews as a source of open-ended knowledge for the community.
+
+But before you head below to read those parts, I would like to convey a big 'Thank you!' to everyone who took part in this, right from making availability and scheduling to providing comprehensive feedback and all the while being extremely communicative. I sincerely appreciate it. Not just the value brought to the table, but also in being great people to talk with in general!
+
+
+## Motivations in contributing
+
+Forms of bringing in people are mostly subjected to one’s interests, which come in various forms:
+
+- Using the software for not just work but for a personal project of interest where they derive usefulness from functions exported by `data.table`.
+
+- Trying to be a part of the `data.table` community, wherein the feeling of being a part of something big and crucial tends to be in play here. Making connections in the open-source community also tends to be an attraction.
+
+- For work reasons, wherein their core goal is to help maintain their own software/package (downstream dependencies) by contributing back.
+
+- For visibility or for employment where `data.table` is a required/preferred tool of trade, and adds to their CV.
+
+- Making one a better programmer. This comes to be not just by contributing directly, but also by learning from others’ contributions.
+
+- Financial gain.
+
+
+## Reasons for not using/adopting
+
+- Low rationale to switch to `data.table` when dealing with small datasets as notable efficiency (in comparison to other R packages that achieve the same functionality) is mostly seen when the data being dealt with is considerably large, or the operations performed on them scale well to see a visible difference.
+
+- On the flipside, there are rare cases where people found `data.table` to be not scalable enough (and instead used DuckDB) for their current datasets, which are increasingly larger than memory (to make operations work on-disk would be something to look out for and implement in the future). An intermediate solution for them here would be to have a syntax translator tool that would translate `data.table` syntax to SQL queries (similar to what `dbplyr` is for `dplyr` commands) as they prefer the `data.table` way to write code and only require things to be more scalable.
+ 
+- Not enough resources online to learn `data.table` in an easy yet detailed manner. This comes in stark contrast to abundant resources available for topics such as `data.frame` and `dplyr` for instance.
+
+- Lack of an integration with `tidyverse`.
```

Comment:
Are there more details here? What kinds of docs online resources did people want to see? What kind of integration would people want?

Reply:
People just mentioned that the resources available online aren't ample or in abundance, as is the case with other topics (such as `dplyr` in terms of packages offering similar functionality, or `data.frame` in terms of the data structures), nothing specific

People wanted data.table to be a part of `tidyverse`